### ISSUE TITLE
Datatable is not updated inside an expansiontemplate

### ISSUE THREAD

**Issue by danicomas** [primefaces/primeng#1775]
**I'm submitting a ...**  (check one with "x")
```
[X] bug report => Search github for a similar issue or PR before submitting
[ ] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primeng/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=35
```

**Plunkr Case (Bug Reports)**
http://plnkr.co/edit/i0S9e9?p=preview

**Current behavior**
If I click on a row of the main datatable, then I click on other row the value of the datatable inside the rowexpansion is not changing.

**Expected behavior**
Value changing.

**Minimal reproduction of the problem with instructions**

**What is the motivation / use case for changing the behavior?**

**Please tell us about your environment:**

* **Angular version:** 2.4.2

* **PrimeNG version:** 1.1.4

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]

 
* **Language:** [all | TypeScript X.X | ES6/7 | ES5]

* **Node (for AoT issues):** `node --version` =

**Comment by danicomas** [comment]
In DataTable RowExpansionLoader (https://github.com/primefaces/primeng/blob/master/components/datatable/datatable.ts#L78) should be implemented by this angular way (https://github.com/angular/angular/issues/13857#issuecomment-273025312)

We only need to destroy the viewRef after detect a change

**Comment by danicomas** [comment]
DataTable RowExpansionLoader (https://github.com/primefaces/primeng/blob/master/components/datatable/datatable.ts#L78) should be implemented by this angular way (angular/angular#13857 (comment)) @cagataycivici

We only need to destroy the viewRef after detect a change

**Comment by cagataycivici** [comment]
Will review for 4.1 final.
